File: 643_maximum_average_subarray_I.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double sum = 0;
        // get first continuous subarray
        for (int i = 0; i < k; i++) {
            sum += (double) nums[i];
        }
        double min = sum;

        // comparing with different length
        int left = 0;
        for (int right = k; right < nums.length; right++) {
            sum += nums[right];
            sum -= nums[left++];
            if (sum > min) {
                min = sum;
            }
        }
        return min / k;
    }
}

File: python/643_maximum_average_subarray_I.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        max_val = curr_sum = sum(nums[:k])
        for i in range(len(nums) - k):
            curr_sum -= nums[i]
            curr_sum += nums[i + k]
            max_val = max(max_val, curr_sum)
        return max_val / k
